GitUI/UserControls/FileStatusList.cs

```diff
@@ -795,13 +795,16 @@ private async Task OpenSubmoduleAsync()
             var submoduleName = SelectedItem.Item.Name;
 
             var status = await SelectedItem.Item.GetSubmoduleStatusAsync().ConfigureAwait(false);
+            var selected = SelectedItem.SecondRevision.ObjectId == ObjectId.WorkTreeId
+                ? SelectedItem.SecondRevision.ObjectId
+                : status.Commit;
 
             var process = new Process
             {
                 StartInfo =
                 {
                     FileName = Application.ExecutablePath,
-                    Arguments = "browse -commit=" + status.Commit,
+                    Arguments = $"browse -commit={selected},{status.OldCommit}",
```

Comment:
This code is identical to RevisionDiffControl.cs L459-474. Should we factor out a new method?

Reply:
I considered that, but settled with aligning the code
Where to put it?

```
        public async Task OpenSubmoduleAsync(FileStatusItem item, IFullPathResolver fullPathResolver)
        {
            var submoduleName = item.Item.Name;

            var status = await item.Item.GetSubmoduleStatusAsync().ConfigureAwait(false);
            var selected = item.SecondRevision.ObjectId == ObjectId.WorkTreeId
                ? item.SecondRevision.ObjectId
                : status.Commit;

            var process = new Process
            {
                StartInfo =
                {
                    FileName = Application.ExecutablePath,
                    Arguments = $"browse -commit={selected},{status.OldCommit}",
                    WorkingDirectory = _fullPathResolver.Resolve(submoduleName.EnsureTrailingPathSeparator())
                }
            };

            process.Start();
        }
```

Reply:
❗  `IFullPathResolver fullPathResolver` must not be passed as an input argument